oi/OITraceCode.cpp

```diff
@@ -61,26 +64,43 @@ class {
  public:
   void initialize() noexcept {
     data.fill(0);
+    numEntries = 0;
   }
 
-  // Adds the pointer to the set.
-  // Returns `true` if the value was newly added,
-  // or `false` if the value was already present.
+  /*
+   * Adds the pointer to the set.
+   * Returns `true` if the value was newly added. `false` may be returned if
+   * the value was already present or if there are no entries left in the
+   * array.
+   */
   bool add(uintptr_t pointer) noexcept {
     __builtin_assume(pointer > 0);
```

Comment:
Could we change this to:
```
if (pointer == 0)
  return false;
```
As a safety feature? It looks to me at the minute that passing `nullptr` would always return true unless the set is full in which case it would return false. It seems like saying always don't track `nullptr` is safer, even though we don't use it like that.

Reply:
Yeah, I forgot about that discussion yesterday! Makes much more sense. I'll change it to that.